### ISSUE TITLE
Documentation of spatial shape constraints for the networks

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/MONAI#6771]
**Is your feature request related to a problem? Please describe.**
follow up of discussion https://github.com/Project-MONAI/MONAI/discussions/6641

would be great to add input spatial size vs model configuration analysis for the networks such as
 swinunetr and dints.

the form of analysis could be detailed tutorials (example https://github.com/Project-MONAI/tutorials/blob/main/modules/UNet_input_size_constrains.ipynb) and/or actual code logic to check the compatibility of the input shape and model configuration.

**Comment by nhhung1810** [comment]
Hi @wyli , I'd like to work on this issue, can you suggest me some related resources that you think is helpful?

**Comment by wyli** [comment]
sure, that's great @nhhung1810, please see:
- how to prepare a pull request https://github.com/Project-MONAI/MONAI/blob/dev/CONTRIBUTING.md
- would be great to revise the related docstring with a bit more details, following this example https://github.com/Project-MONAI/MONAI/blob/6f9cf6b1c1ffade7af37f0b2bed1b9b517711a00/monai/networks/nets/unet.py#L32-L54
- the docstring format https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html
- the docstring style https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings

**Comment by nhhung1810** [comment]
> all the networks such as swinunetr and dints
Hi @wyli , I think that you refer to all networks under `monai/networks/nets`? If such, can you suggest some high priority networks first? 

Also, if the number of networks need to be done is a lot, I'll split up into small PRs for each networks, do you fine with that?

**Comment by wyli** [comment]
Yes @nhhung1810, separate PRs would be great, high priority ones are
- [ ] https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/basic_unetplusplus.py
- [ ] https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/segresnet_ds.py
- [ ] https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/swin_unetr.py
- [ ] https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/dints.py
please feel free to start with simple PRs.. we can use this ticket to track the progress

**Comment by nhhung1810** [comment]
Sure, I will start with UNET++ first

**Comment by nhhung1810** [comment]
Hi @wyli I have create a PR here https://github.com/Project-MONAI/tutorials/pull/1476, mentioned in the above Issue https://github.com/Project-MONAI/tutorials/issues/1475

**Comment by vikashg** [comment]
Good job @nhhung1810

**Comment by vikashg** [comment]
@nhhung1810 is it completed?

**Comment by nhhung1810** [comment]
Oh it's complete @vikashg, it's in the PR above, for 1 model (UNET++) only. I haven't worked on this since then @vikashg

**Comment by Sheetala28** [comment]
"Hi @wyli, I'm a new contributor and I'd like to take on this issue. I'll start by reviewing the UNet size constraints notebook mentioned in the description."
